### PR TITLE
AVM2: move cached namespaces to a new `CommonNamespaces` struct

### PR DESCRIPTION
**Pull request by moulins**
Extracted from #17876; this mirrors #17842.

### CODE REVIEW

File: core/src/avm2/globals.rs [hunk 1]
```diff
@@ -557,30 +555,28 @@ pub fn load_player_globals<'gc>(
...
-    let public_ns = activation.avm2().public_namespace_base_version;
-    let vector_public_ns = activation.avm2().vector_public_namespace;
-    let vector_internal_ns = activation.avm2().vector_internal_namespace;
-
     // Unfortunately we need to specify the global traits manually, at least until
     // all the builtin classes are defined in AS.
     let mut global_traits = Vec::new();
 
+    let namespaces = activation.avm2().namespaces;
+    #[rustfmt::skip]
```
Comment (Lord-McSweeney) [inline core/src/avm2/globals.rs:563]: Can we keep the above
```
    let public_ns = activation.avm2().public_namespace_base_version;
    let vector_public_ns = activation.avm2().vector_public_namespace;
    let vector_internal_ns = activation.avm2().vector_internal_namespace;
```
and avoid skipping fmt here?

File: core/src/avm2/namespace.rs [hunk 2]
```diff
@@ -336,3 +336,77 @@ impl<'gc> Namespace<'gc> {
...
+
+    // These are required to facilitate shared access between Rust and AS.
+    pub(super) flash_display_internal: Namespace<'gc>,
+    pub(super) flash_utils_internal: Namespace<'gc>,
+    pub(super) flash_geom_internal: Namespace<'gc>,
+    pub(super) flash_events_internal: Namespace<'gc>,
+    pub(super) flash_text_engine_internal: Namespace<'gc>,
+    pub(super) flash_net_internal: Namespace<'gc>,
+
+    pub(super) __ruffle__: Namespace<'gc>,
```
Comment (Lord-McSweeney) [inline core/src/avm2/namespace.rs:360]: Can this be called `ruffle_internal` instead?

**Review by Lord-McSweeney — APPROVED** [review]
LGTM besides nits

File: core/src/avm2/namespace.rs (hunk 2 above)
Comment (moulins) [reply core/src/avm2/namespace.rs:360]: This namespace is actually public, because it's purpose for Ruffle-specific functions we want to call in tests.

**Pull request merged**